src/TestFramework/Coverage/XmlReport/TestLocator.php

```diff
@@ -96,11 +97,19 @@ private function getTestsForFunctionSignature(NodeLineRangeData $lineRange): ite
      */
     private function getTestsForLineRange(NodeLineRangeData $lineRange): iterable
     {
+        // same test can cover more than 1 line. To avoid many duplications, we need to return unique tests after
+        // accumulating them by each line from the range
+        $uniqueTestLocations = [];
+
         foreach ($lineRange->range as $line) {
             if (array_key_exists($line, $this->testLocations->getTestsLocationsBySourceLine())) {
-                yield from $this->testLocations->getTestsLocationsBySourceLine()[$line];
+                foreach ($this->testLocations->getTestsLocationsBySourceLine()[$line] as $testLocation) {
```

Comment:
can't we get rid of `if` here? and do smth like:
```
foreach ($this->testLocations->getTestsLocationsBySourceLine()[$line] ?? [] as $testLocation) {
   //do smth
}
```

Reply:
fixed here https://github.com/infection/infection/commit/6a6eb17b475786b59adee5572f16d579a5eb4e63